tfe/resource_tfe_notification_configuration_test.go

```diff
@@ -0,0 +1,436 @@
+				ResourceName:            "tfe_notification_configuration.foobar",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"token", "workspace_external_id"},
+			},
+
+			{
+				Config: testAccTFENotificationConfiguration_update,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTFENotificationConfigurationExists(
+						"tfe_notification_configuration.foobar", notificationConfiguration),
+					testAccCheckTFENotificationConfigurationAttributesUpdate(notificationConfiguration),
+					resource.TestCheckResourceAttr(
+						"tfe_notification_configuration.foobar", "destination_type", "generic"),
+					resource.TestCheckResourceAttr(
+						"tfe_notification_configuration.foobar", "enabled", "true"),
+					resource.TestCheckResourceAttr(
+						"tfe_notification_configuration.foobar", "name", "notification_update"),
+					resource.TestCheckResourceAttr(
+						"tfe_notification_configuration.foobar", "token", "1234567890_update"),
+					// Just test the number of items in triggers
+					// Values in triggers attribute are tested by testCheckTFENotificationConfigurationAttributesUpdate
+					resource.TestCheckResourceAttr(
+						"tfe_notification_configuration.foobar", "triggers.#", "2"),
+					resource.TestCheckResourceAttr(
+						"tfe_notification_configuration.foobar", "url", "http://example.com/?update=true"),
+				),
+			},
+
+			{
+				ResourceName:            "tfe_notification_configuration.foobar",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"token", "workspace_external_id"},
+			},
+		},
+	})
+}
+
+func TestAccTFENotificationConfiguration_slackWithToken(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckTFENotificationConfigurationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccTFENotificationConfiguration_slackWithToken,
+				ExpectError: regexp.MustCompile(`token cannot be set with destination_type of slack`),
+			},
+		},
+	})
+}
+
+func TestAccTFENotificationConfiguration_wrongDestinationType(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckTFENotificationConfigurationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccTFENotificationConfiguration_wrongDestinationType,
+				ExpectError: regexp.MustCompile(``),
+			},
+		},
+	})
+}
+
+func TestAccTFENotificationConfiguration_wrongTriggers(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckTFENotificationConfigurationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccTFENotificationConfiguration_wrongTriggers,
+				ExpectError: regexp.MustCompile(``),
```

Comment:
Same question here ☝️

Reply:
Removed, I think your point about other tests not including this is a good one.